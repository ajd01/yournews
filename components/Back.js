import React, { useContext } from 'react'
import { View, StyleSheet, Dimensions } from 'react-native'
import Icon from './Icon'
import { Context } from '../store'

export default function Back () {
  const { dispatch } = useContext(Context)

  return (
    <View
      style={styles.Back} onClick={() => {
        dispatch({
          type: 'unselect'
        })
      }}
    >
      <Icon name='arrow-left' type='font-awesome' color='#f50' />
    </View>
  )
}

const styles = StyleSheet.create({
  Back: {
    textAlign: 'left',
    width: Dimensions.get('window').width,
    position: 'relative',
    flexDirection: 'row',
    margin: '20px',
    zIndex: 99
  }
})
