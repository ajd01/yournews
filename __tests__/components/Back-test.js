import React from 'react'
import Back from '../../components/Back'
import renderer from 'react-test-renderer'
import {
  Context,
  InitialState
} from '../../store'

jest.mock('../../components/Icon', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    }
  }
})

const value = {
  dispatch: jest.fn(),
  store: InitialState
}

describe('Back componente', () => {
  it('renders screen', () => {
    const tree = renderer.create(
      <Context.Provider value={value}>
        <Back />
      </Context.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
