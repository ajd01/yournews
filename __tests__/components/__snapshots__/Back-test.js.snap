// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back componente renders screen 1`] = `
<View
  onClick={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "margin": "20px",
      "position": "relative",
      "textAlign": "left",
      "width": 750,
      "zIndex": 99,
    }
  }
>
  <div />
</View>
`;
